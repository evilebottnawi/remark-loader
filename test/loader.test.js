import path from "path";

import RemarkHTML from "remark-html";
import RemarkGFM from "remark-gfm";
import RemarkBookmarks from "remark-bookmarks";
import RemarkFrontmatter from "remark-frontmatter";

import {
  compile,
  getExecutedCode,
  getCompiler,
  getErrors,
  getWarnings,
} from "./helpers";

describe("loader", () => {
  it("should work markdown to markdown", async () => {
    const compiler = getCompiler("simple.js");
    const stats = await compile(compiler);
    const codeFromBundle = getExecutedCode("main.bundle.js", compiler, stats);

    expect(codeFromBundle.md).toMatchSnapshot("md");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should work markdown to html", async () => {
    const compiler = getCompiler(
      "simple.js",
      {},
      {
        module: {
          rules: [
            {
              test: /\.md$/i,
              rules: [
                {
                  loader: require.resolve("./helpers/testLoader"),
                },
                {
                  loader: "html-loader",
                },
                {
                  loader: path.resolve(__dirname, "../src"),
                  options: {
                    remarkOptions: {
                      plugins: [RemarkHTML],
                    },
                  },
                },
              ],
            },
          ],
        },
      }
    );
    const stats = await compile(compiler);
    const codeFromBundle = getExecutedCode("main.bundle.js", compiler, stats);

    expect(codeFromBundle.md).toMatchSnapshot("md");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should work with the 'remark-gfm' plugin", async () => {
    const compiler = getCompiler("simple.js", {
      remarkOptions: {
        plugins: [RemarkGFM],
      },
    });
    const stats = await compile(compiler);
    const codeFromBundle = getExecutedCode("main.bundle.js", compiler, stats);

    expect(codeFromBundle.md).toMatchSnapshot("md");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should work if plugins are array", async () => {
    const compiler = getCompiler("multipleArgs.js", {
      remarkOptions: {
        plugins: [
          [
            RemarkBookmarks,
            {
              bookmarks: {
                npm: "https://npmjs.com/package/remark-bookmarks",
              },
            },
          ],
        ],
      },
    });
    const stats = await compile(compiler);
    const codeFromBundle = getExecutedCode("main.bundle.js", compiler, stats);

    expect(codeFromBundle.md).toMatchSnapshot("md");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should not remove frontmatter", async () => {
    const compiler = getCompiler("multipleArgs.js", {
      removeFrontMatter: false,
      remarkOptions: {
        plugins: [RemarkFrontmatter],
      },
    });
    const stats = await compile(compiler);
    const codeFromBundle = getExecutedCode("main.bundle.js", compiler, stats);

    expect(codeFromBundle.md).toMatchSnapshot("md");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should work with the 'settings' option", async () => {
    const compiler = getCompiler("multipleArgs.js", {
      remarkOptions: {
        settings: {
          bullet: "+",
          listItemIndent: "1",
        },
      },
    });
    const stats = await compile(compiler);
    const codeFromBundle = getExecutedCode("main.bundle.js", compiler, stats);

    expect(codeFromBundle.md).toMatchSnapshot("md");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should work with the 'settings' option in plugins", async () => {
    const compiler = getCompiler("multipleArgs.js", {
      remarkOptions: {
        plugins: [
          {
            settings: {
              bullet: "+",
              listItemIndent: "1",
            },
          },
        ],
      },
    });
    const stats = await compile(compiler);
    const codeFromBundle = getExecutedCode("main.bundle.js", compiler, stats);

    expect(codeFromBundle.md).toMatchSnapshot("md");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should work with the 'data' option", async () => {
    let alpha;
    let charlie;

    function extractDataPlugin() {
      alpha = this.data("alpha");
      charlie = this.data("charlie");
    }

    const compiler = getCompiler("multipleArgs.js", {
      remarkOptions: {
        plugins: [extractDataPlugin],
        data: {
          alpha: "bravo",
          charlie: "delta",
        },
      },
    });
    const stats = await compile(compiler);
    const codeFromBundle = getExecutedCode("main.bundle.js", compiler, stats);

    expect(alpha).toEqual("bravo");
    expect(charlie).toEqual("delta");
    expect(codeFromBundle.md).toMatchSnapshot("md");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should throw error #1", async () => {
    const compiler = getCompiler("multipleArgs.js", {
      removeFrontMatter: false,
      remarkOptions: {
        plugins: [[RemarkFrontmatter, { marker: "*" }]],
      },
    });

    const stats = await compile(compiler);

    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should throw error #2", async () => {
    const errorGenerationPlugin = () => () => {
      throw new Error("Error");
    };

    const compiler = getCompiler("multipleArgs.js", {
      removeFrontMatter: false,
      remarkOptions: {
        plugins: [errorGenerationPlugin],
      },
    });

    const stats = await compile(compiler);

    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });
});
