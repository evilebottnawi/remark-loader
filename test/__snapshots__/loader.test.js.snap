// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader should not remove frontmatter: errors 1`] = `Array []`;

exports[`loader should not remove frontmatter: md 1`] = `
"---
title: Just hack'n
description: Nothing to see here
meta: some meta data
---

# My header

Test content.

\`remark-bookmarks\` is on \\\\[npm]\\\\[]!

Example list:

*   item 1
*   item 2
*   item 3
"
`;

exports[`loader should not remove frontmatter: warnings 1`] = `Array []`;

exports[`loader should throw error #1: errors 1`] = `
Array [
  "ModuleBuildError: Module build failed (from \`replaced original path\`):
Error: Missing \`type\` in matter \`{\\"marker\\":\\"*\\"}\`",
]
`;

exports[`loader should throw error #1: warnings 1`] = `Array []`;

exports[`loader should throw error #2: errors 1`] = `
Array [
  "ModuleBuildError: Module build failed (from \`replaced original path\`):
NonErrorEmittedError: (Emitted value instead of an instance of Error) Error: Error",
]
`;

exports[`loader should throw error #2: warnings 1`] = `Array []`;

exports[`loader should work if plugins are array: errors 1`] = `Array []`;

exports[`loader should work if plugins are array: md 1`] = `
"# My header

Test content.

\`remark-bookmarks\` is on \\\\[npm]\\\\[]!

Example list:

*   item 1
*   item 2
*   item 3
"
`;

exports[`loader should work if plugins are array: warnings 1`] = `Array []`;

exports[`loader should work markdown to html: errors 1`] = `Array []`;

exports[`loader should work markdown to html: md 1`] = `
"// Module
var code = \\"<h1>h1 Heading 8-)</h1>\\\\n<h2>h2 Heading</h2>\\\\n<h3>h3 Heading</h3>\\\\n<h4>h4 Heading</h4>\\\\n<h5>h5 Heading</h5>\\\\n<h6>h6 Heading</h6>\\\\n<h2>Horizontal Rules</h2>\\\\n<hr>\\\\n<hr>\\\\n<hr>\\\\n<h2>Typographic replacements</h2>\\\\n<p>Enable typographer option to see result.</p>\\\\n<p>(c) (C) (r) (R) (tm) (TM) (p) (P) +-</p>\\\\n<p>test.. test... test..... test?..... test!....</p>\\\\n<p>!!!!!! ???? ,,  -- ---</p>\\\\n<p>\\\\\\"Smartypants, double quotes\\\\\\" and 'single quotes'</p>\\\\n<h2>Emphasis</h2>\\\\n<p><strong>This is bold text</strong></p>\\\\n<p><strong>This is bold text</strong></p>\\\\n<p><em>This is italic text</em></p>\\\\n<p><em>This is italic text</em></p>\\\\n<p>~~Strikethrough~~</p>\\\\n<h2>Blockquotes</h2>\\\\n<blockquote>\\\\n<p>Blockquotes can also be nested...</p>\\\\n<blockquote>\\\\n<p>...by using additional greater-than signs right next to each other...</p>\\\\n<blockquote>\\\\n<p>...or with spaces between arrows.</p>\\\\n</blockquote>\\\\n</blockquote>\\\\n</blockquote>\\\\n<h2>Lists</h2>\\\\n<p>Unordered</p>\\\\n<ul>\\\\n<li>Create a list by starting a line with <code>+</code>, <code>-</code>, or <code>*</code></li>\\\\n<li>Sub-lists are made by indenting 2 spaces:\\\\n<ul>\\\\n<li>Marker character change forces new list start:\\\\n<ul>\\\\n<li>Ac tristique libero volutpat at</li>\\\\n</ul>\\\\n<ul>\\\\n<li>Facilisis in pretium nisl aliquet</li>\\\\n</ul>\\\\n<ul>\\\\n<li>Nulla volutpat aliquam velit</li>\\\\n</ul>\\\\n</li>\\\\n</ul>\\\\n</li>\\\\n<li>Very easy!</li>\\\\n</ul>\\\\n<p>Ordered</p>\\\\n<ol>\\\\n<li>\\\\n<p>Lorem ipsum dolor sit amet</p>\\\\n</li>\\\\n<li>\\\\n<p>Consectetur adipiscing elit</p>\\\\n</li>\\\\n<li>\\\\n<p>Integer molestie lorem at massa</p>\\\\n</li>\\\\n<li>\\\\n<p>You can use sequential numbers...</p>\\\\n</li>\\\\n<li>\\\\n<p>...or keep all the numbers as <code>1.</code></p>\\\\n</li>\\\\n</ol>\\\\n<p>Start numbering with offset:</p>\\\\n<ol start=\\\\\\"57\\\\\\">\\\\n<li>foo</li>\\\\n<li>bar</li>\\\\n</ol>\\\\n<h2>Code</h2>\\\\n<p>Inline <code>code</code></p>\\\\n<p>Indented code</p>\\\\n<pre><code>// Some comments\\\\nline 1 of code\\\\nline 2 of code\\\\nline 3 of code\\\\n</code></pre>\\\\n<p>Block code \\\\\\"fences\\\\\\"</p>\\\\n<pre><code>Sample text here...\\\\n</code></pre>\\\\n<p>Syntax highlighting</p>\\\\n<pre><code class=\\\\\\"language-js\\\\\\">var foo = function (bar) {\\\\n  return bar++;\\\\n};\\\\n\\\\nconsole.log(foo(5));\\\\n</code></pre>\\\\n<h2>Tables</h2>\\\\n<p>| Option | Description |\\\\n| ------ | ----------- |\\\\n| data   | path to data files to supply the data that will be passed into templates. |\\\\n| engine | engine to be used for processing templates. Handlebars is the default. |\\\\n| ext    | extension to be used for dest files. |</p>\\\\n<p>Right aligned columns</p>\\\\n<p>| Option | Description |\\\\n| ------:| -----------:|\\\\n| data   | path to data files to supply the data that will be passed into templates. |\\\\n| engine | engine to be used for processing templates. Handlebars is the default. |\\\\n| ext    | extension to be used for dest files. |</p>\\\\n<h2>Links</h2>\\\\n<p><a href=\\\\\\"http://dev.nodeca.com\\\\\\">link text</a></p>\\\\n<p><a href=\\\\\\"http://nodeca.github.io/pica/demo/\\\\\\" title=\\\\\\"title text!\\\\\\">link with title</a></p>\\\\n<p>Autoconverted link https://github.com/nodeca/pica (enable linkify to see)</p>\\\\n<h2>Images</h2>\\\\n<p><img src=\\\\\\"https://octodex.github.com/images/minion.png\\\\\\" alt=\\\\\\"Minion\\\\\\">\\\\n<img src=\\\\\\"https://octodex.github.com/images/stormtroopocat.jpg\\\\\\" alt=\\\\\\"Stormtroopocat\\\\\\" title=\\\\\\"The Stormtroopocat\\\\\\"></p>\\\\n<p>Like links, Images also have a footnote style syntax</p>\\\\n<p><img src=\\\\\\"https://octodex.github.com/images/dojocat.jpg\\\\\\" alt=\\\\\\"Alt text\\\\\\" title=\\\\\\"The Dojocat\\\\\\"></p>\\\\n<p>With a reference later in the document defining the URL location:</p>\\\\n<h2>Plugins</h2>\\\\n<p>The killer feature of <code>markdown-it</code> is very effective support of\\\\n<a href=\\\\\\"https://www.npmjs.org/browse/keyword/markdown-it-plugin\\\\\\">syntax plugins</a>.</p>\\\\n<h3><a href=\\\\\\"https://github.com/markdown-it/markdown-it-emoji\\\\\\">Emojies</a></h3>\\\\n<blockquote>\\\\n<p>Classic markup: :wink: :crush: :cry: :tear: :laughing: :yum:</p>\\\\n<p>Shortcuts (emoticons): :-) :-( 8-) ;)</p>\\\\n</blockquote>\\\\n<p>see <a href=\\\\\\"https://github.com/markdown-it/markdown-it-emoji#change-output\\\\\\">how to change output</a> with twemoji.</p>\\\\n<h3><a href=\\\\\\"https://github.com/markdown-it/markdown-it-sub\\\\\\">Subscript</a> / <a href=\\\\\\"https://github.com/markdown-it/markdown-it-sup\\\\\\">Superscript</a></h3>\\\\n<ul>\\\\n<li>19^th^</li>\\\\n<li>H~2~O</li>\\\\n</ul>\\\\n<h3><a href=\\\\\\"https://github.com/markdown-it/markdown-it-ins\\\\\\">&#x3C;ins></a></h3>\\\\n<p>++Inserted text++</p>\\\\n<h3><a href=\\\\\\"https://github.com/markdown-it/markdown-it-mark\\\\\\">&#x3C;mark></a></h3>\\\\n<p>==Marked text==</p>\\\\n<h3><a href=\\\\\\"https://github.com/markdown-it/markdown-it-footnote\\\\\\">Footnotes</a></h3>\\\\n<p>Footnote 1 link[^first].</p>\\\\n<p>Footnote 2 link[^second].</p>\\\\n<p>Inline footnote^[Text of inline footnote] definition.</p>\\\\n<p>Duplicated footnote reference[^second].</p>\\\\n<p>[^first]: Footnote <strong>can have markup</strong></p>\\\\n<pre><code>and multiple paragraphs.\\\\n</code></pre>\\\\n<p>[^second]: Footnote text.</p>\\\\n<h3><a href=\\\\\\"https://github.com/markdown-it/markdown-it-deflist\\\\\\">Definition lists</a></h3>\\\\n<p>Term 1</p>\\\\n<p>:   Definition 1\\\\nwith lazy continuation.</p>\\\\n<p>Term 2 with <em>inline markup</em></p>\\\\n<p>:   Definition 2</p>\\\\n<pre><code>    { some code, part of Definition 2 }\\\\n\\\\nThird paragraph of definition 2.\\\\n</code></pre>\\\\n<p><em>Compact style:</em></p>\\\\n<p>Term 1\\\\n~ Definition 1</p>\\\\n<p>Term 2\\\\n~ Definition 2a\\\\n~ Definition 2b</p>\\\\n<h3><a href=\\\\\\"https://github.com/markdown-it/markdown-it-abbr\\\\\\">Abbreviations</a></h3>\\\\n<p>This is HTML abbreviation example.</p>\\\\n<p>It converts \\\\\\"HTML\\\\\\", but keep intact partial entries like \\\\\\"xxxHTMLyyy\\\\\\" and so on.</p>\\\\n<p>*[HTML]: Hyper Text Markup Language</p>\\\\n<h3><a href=\\\\\\"https://github.com/markdown-it/markdown-it-container\\\\\\">Custom containers</a></h3>\\\\n<p>::: warning\\\\n<em>here be dragons</em>\\\\n:::</p>\\\\n\\";
// Exports
module.exports = code;"
`;

exports[`loader should work markdown to html: warnings 1`] = `Array []`;

exports[`loader should work markdown to markdown: errors 1`] = `Array []`;

exports[`loader should work markdown to markdown: md 1`] = `
"# h1 Heading 8-)

## h2 Heading

### h3 Heading

#### h4 Heading

##### h5 Heading

###### h6 Heading

## Horizontal Rules

***

***

***

## Typographic replacements

Enable typographer option to see result.

(c) (C) (r) (R) (tm) (TM) (p) (P) +-

test.. test... test..... test?..... test!....

!!!!!! ???? ,,  -- ---

\\"Smartypants, double quotes\\" and 'single quotes'

## Emphasis

**This is bold text**

**This is bold text**

*This is italic text*

*This is italic text*

\\\\~~Strikethrough~~

## Blockquotes

> Blockquotes can also be nested...
>
> > ...by using additional greater-than signs right next to each other...
> >
> > > ...or with spaces between arrows.

## Lists

Unordered

*   Create a list by starting a line with \`+\`, \`-\`, or \`*\`
*   Sub-lists are made by indenting 2 spaces:
    *   Marker character change forces new list start:
        *   Ac tristique libero volutpat at

        <!---->

        *   Facilisis in pretium nisl aliquet

        <!---->

        *   Nulla volutpat aliquam velit
*   Very easy!

Ordered

1.  Lorem ipsum dolor sit amet

2.  Consectetur adipiscing elit

3.  Integer molestie lorem at massa

4.  You can use sequential numbers...

5.  ...or keep all the numbers as \`1.\`

Start numbering with offset:

57. foo
58. bar

## Code

Inline \`code\`

Indented code

    // Some comments
    line 1 of code
    line 2 of code
    line 3 of code

Block code \\"fences\\"

    Sample text here...

Syntax highlighting

\`\`\`js
var foo = function (bar) {
  return bar++;
};

console.log(foo(5));
\`\`\`

## Tables

| Option | Description |
| ------ | ----------- |
| data   | path to data files to supply the data that will be passed into templates. |
| engine | engine to be used for processing templates. Handlebars is the default. |
| ext    | extension to be used for dest files. |

Right aligned columns

| Option | Description |
| ------:| -----------:|
| data   | path to data files to supply the data that will be passed into templates. |
| engine | engine to be used for processing templates. Handlebars is the default. |
| ext    | extension to be used for dest files. |

## Links

[link text](http://dev.nodeca.com)

[link with title](http://nodeca.github.io/pica/demo/ \\"title text!\\")

Autoconverted link https://github.com/nodeca/pica (enable linkify to see)

## Images

![Minion](https://octodex.github.com/images/minion.png)
![Stormtroopocat](https://octodex.github.com/images/stormtroopocat.jpg \\"The Stormtroopocat\\")

Like links, Images also have a footnote style syntax

![Alt text][id]

With a reference later in the document defining the URL location:

[id]: https://octodex.github.com/images/dojocat.jpg \\"The Dojocat\\"

## Plugins

The killer feature of \`markdown-it\` is very effective support of
[syntax plugins](https://www.npmjs.org/browse/keyword/markdown-it-plugin).

### [Emojies](https://github.com/markdown-it/markdown-it-emoji)

> Classic markup: :wink: :crush: :cry: :tear: :laughing: :yum:
>
> Shortcuts (emoticons): :-) :-( 8-) ;)

see [how to change output](https://github.com/markdown-it/markdown-it-emoji#change-output) with twemoji.

### [Subscript](https://github.com/markdown-it/markdown-it-sub) / [Superscript](https://github.com/markdown-it/markdown-it-sup)

*   19^th^
*   H~2~O

### [\\\\<ins>](https://github.com/markdown-it/markdown-it-ins)

\\\\++Inserted text++

### [\\\\<mark>](https://github.com/markdown-it/markdown-it-mark)

\\\\==Marked text==

### [Footnotes](https://github.com/markdown-it/markdown-it-footnote)

Footnote 1 link\\\\[^first].

Footnote 2 link\\\\[^second].

Inline footnote^\\\\[Text of inline footnote] definition.

Duplicated footnote reference\\\\[^second].

\\\\[^first]: Footnote **can have markup**

    and multiple paragraphs.

\\\\[^second]: Footnote text.

### [Definition lists](https://github.com/markdown-it/markdown-it-deflist)

Term 1

:   Definition 1
with lazy continuation.

Term 2 with *inline markup*

:   Definition 2

        { some code, part of Definition 2 }

    Third paragraph of definition 2.

*Compact style:*

Term 1
\\\\~ Definition 1

Term 2
\\\\~ Definition 2a
\\\\~ Definition 2b

### [Abbreviations](https://github.com/markdown-it/markdown-it-abbr)

This is HTML abbreviation example.

It converts \\"HTML\\", but keep intact partial entries like \\"xxxHTMLyyy\\" and so on.

\\\\*\\\\[HTML]: Hyper Text Markup Language

### [Custom containers](https://github.com/markdown-it/markdown-it-container)

::: warning
*here be dragons*
:::
"
`;

exports[`loader should work markdown to markdown: warnings 1`] = `Array []`;

exports[`loader should work with the 'data' option: errors 1`] = `Array []`;

exports[`loader should work with the 'data' option: md 1`] = `
"# My header

Test content.

\`remark-bookmarks\` is on \\\\[npm]\\\\[]!

Example list:

*   item 1
*   item 2
*   item 3
"
`;

exports[`loader should work with the 'data' option: warnings 1`] = `Array []`;

exports[`loader should work with the 'remark-gfm' plugin: errors 1`] = `Array []`;

exports[`loader should work with the 'remark-gfm' plugin: md 1`] = `
"# h1 Heading 8-)

## h2 Heading

### h3 Heading

#### h4 Heading

##### h5 Heading

###### h6 Heading

## Horizontal Rules

***

***

***

## Typographic replacements

Enable typographer option to see result.

(c) (C) (r) (R) (tm) (TM) (p) (P) +-

test.. test... test..... test?..... test!....

!!!!!! ???? ,,  -- ---

\\"Smartypants, double quotes\\" and 'single quotes'

## Emphasis

**This is bold text**

**This is bold text**

*This is italic text*

*This is italic text*

~~Strikethrough~~

## Blockquotes

> Blockquotes can also be nested...
>
> > ...by using additional greater-than signs right next to each other...
> >
> > > ...or with spaces between arrows.

## Lists

Unordered

*   Create a list by starting a line with \`+\`, \`-\`, or \`*\`
*   Sub-lists are made by indenting 2 spaces:
    *   Marker character change forces new list start:
        *   Ac tristique libero volutpat at

        <!---->

        *   Facilisis in pretium nisl aliquet

        <!---->

        *   Nulla volutpat aliquam velit
*   Very easy!

Ordered

1.  Lorem ipsum dolor sit amet

2.  Consectetur adipiscing elit

3.  Integer molestie lorem at massa

4.  You can use sequential numbers...

5.  ...or keep all the numbers as \`1.\`

Start numbering with offset:

57. foo
58. bar

## Code

Inline \`code\`

Indented code

    // Some comments
    line 1 of code
    line 2 of code
    line 3 of code

Block code \\"fences\\"

    Sample text here...

Syntax highlighting

\`\`\`js
var foo = function (bar) {
  return bar++;
};

console.log(foo(5));
\`\`\`

## Tables

| Option | Description                                                               |
| ------ | ------------------------------------------------------------------------- |
| data   | path to data files to supply the data that will be passed into templates. |
| engine | engine to be used for processing templates. Handlebars is the default.    |
| ext    | extension to be used for dest files.                                      |

Right aligned columns

| Option |                                                               Description |
| -----: | ------------------------------------------------------------------------: |
|   data | path to data files to supply the data that will be passed into templates. |
| engine |    engine to be used for processing templates. Handlebars is the default. |
|    ext |                                      extension to be used for dest files. |

## Links

[link text](http://dev.nodeca.com)

[link with title](http://nodeca.github.io/pica/demo/ \\"title text!\\")

Autoconverted link <https://github.com/nodeca/pica> (enable linkify to see)

## Images

![Minion](https://octodex.github.com/images/minion.png)
![Stormtroopocat](https://octodex.github.com/images/stormtroopocat.jpg \\"The Stormtroopocat\\")

Like links, Images also have a footnote style syntax

![Alt text][id]

With a reference later in the document defining the URL location:

[id]: https://octodex.github.com/images/dojocat.jpg \\"The Dojocat\\"

## Plugins

The killer feature of \`markdown-it\` is very effective support of
[syntax plugins](https://www.npmjs.org/browse/keyword/markdown-it-plugin).

### [Emojies](https://github.com/markdown-it/markdown-it-emoji)

> Classic markup: :wink: :crush: :cry: :tear: :laughing: :yum:
>
> Shortcuts (emoticons): :-) :-( 8-) ;)

see [how to change output](https://github.com/markdown-it/markdown-it-emoji#change-output) with twemoji.

### [Subscript](https://github.com/markdown-it/markdown-it-sub) / [Superscript](https://github.com/markdown-it/markdown-it-sup)

*   19^th^
*   H~~2~~O

### [\\\\<ins>](https://github.com/markdown-it/markdown-it-ins)

\\\\++Inserted text++

### [\\\\<mark>](https://github.com/markdown-it/markdown-it-mark)

\\\\==Marked text==

### [Footnotes](https://github.com/markdown-it/markdown-it-footnote)

Footnote 1 link\\\\[^first].

Footnote 2 link\\\\[^second].

Inline footnote^\\\\[Text of inline footnote] definition.

Duplicated footnote reference\\\\[^second].

\\\\[^first]: Footnote **can have markup**

    and multiple paragraphs.

\\\\[^second]: Footnote text.

### [Definition lists](https://github.com/markdown-it/markdown-it-deflist)

Term 1

:   Definition 1
with lazy continuation.

Term 2 with *inline markup*

:   Definition 2

        { some code, part of Definition 2 }

    Third paragraph of definition 2.

*Compact style:*

Term 1
\\\\~ Definition 1

Term 2
\\\\~ Definition 2a
\\\\~ Definition 2b

### [Abbreviations](https://github.com/markdown-it/markdown-it-abbr)

This is HTML abbreviation example.

It converts \\"HTML\\", but keep intact partial entries like \\"xxxHTMLyyy\\" and so on.

\\\\*\\\\[HTML]: Hyper Text Markup Language

### [Custom containers](https://github.com/markdown-it/markdown-it-container)

::: warning
*here be dragons*
:::
"
`;

exports[`loader should work with the 'remark-gfm' plugin: warnings 1`] = `Array []`;

exports[`loader should work with the 'settings' option in plugins: errors 1`] = `Array []`;

exports[`loader should work with the 'settings' option in plugins: md 1`] = `
"# My header

Test content.

\`remark-bookmarks\` is on \\\\[npm]\\\\[]!

Example list:

+ item 1
+ item 2
+ item 3
"
`;

exports[`loader should work with the 'settings' option in plugins: warnings 1`] = `Array []`;

exports[`loader should work with the 'settings' option: errors 1`] = `Array []`;

exports[`loader should work with the 'settings' option: md 1`] = `
"# My header

Test content.

\`remark-bookmarks\` is on \\\\[npm]\\\\[]!

Example list:

+ item 1
+ item 2
+ item 3
"
`;

exports[`loader should work with the 'settings' option: warnings 1`] = `Array []`;
